### DISCUSSION TITLE
Planning failed: Terraform encountered an error while generating this plan

### DISCUSSION THREAD

**Discussion by umar-shabbir** [eclipse-theia/theia-cloud#274] [Q&A]
I am trying to follow [getting started with Minikube](https://github.com/eclipsesource/theia-cloud/blob/main/doc/docs/platforms/Minikube.md#getting-started-with-minikube) instructions and I get following error when running `terraform plan`

**Planning failed. Terraform encountered an error while generating this plan.**

```
╷
│ Error: Plugin did not respond
│
│   with module.cluster.minikube_cluster.cluster,
│   on ..\..\modules\cluster_creation\minikube\main.tf line 26, in resource "minikube_cluster" "cluster":
│   26: resource "minikube_cluster" "cluster" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
```

**Steps tried:**

- I have followed the [Hello Minikube](https://kubernetes.io/docs/tutorials/hello-minikube/) tutorial before starting Theia Cloud. Steps in that tutorial worked fine for me. 
- Tried setting **KUBECONFIG** and **KUBE_CONFIG_PATH** environment variables
- Tried deleting all **terraform lock** files and running `terraform init`, `terraform plan` again
- Tried `minikube delete` and following all the steps again
- Uninstall **minikube**, remove **.kube** from user profile. Reinstall and try all steps again

Any suggestions for above issue are appreciated. If someone has a working sample of Theia Cloud with Minikube tested on Windows 11, that can be helpful too.

**Additional Information:**
Platform: Windows 11
Terraform: 1.7.0
Minikube: 1.32.0
VirtualBox: 7.0.12

**Comment by jfaltermeier** [comment]
Hi, I am usually working in Ubuntu but I got my hands on a windows 10 machine and tested the setup there. Sorry that I can't test on windows 11. 

On my first run I got the same error as you posted above. 
I then ran `git clean -xfd` (to delete any local terraform state) and `minikube delete` (to delete the failed minikube VM). I think this is the same as the steps you tried above.
In order to get more logs from terraform I ran `export TF_LOG=true` (using the git bash on windows). This will log the trace logs to the console. 
Without any other change the setup worked now. (Or nearly, I got two more error, because I did not have `kubectl` and `ruby` installed, but with those the setup worked):

![TheiaCloudWindows](https://github.com/eclipsesource/theia-cloud/assets/5889696/e60c840a-df42-481d-bcd1-d2c5a069b2c9)

This was not run as an administrator, I just had to accept some changes to the firewall rules while the cluster was created by terraform. 

So maybe retrying again on a clean state will help you. Otherwise the terraform logs may give further hints.

**Comment by umar-shabbir** [comment]
@jfaltermeier thanks for trying it out and sharing the result.
I did try on a clean setup by following what you have described above. Though it solved the previous error but now I am getting another error.
`Error: Kubernetes cluster unreachable: invalid configuration: [client-cert-data and client-cert are both specified for minikube. client-cert-data will override., client-key-data and client-key are both specified for minikube; client-key-data will override]`

From the logs I found that this same error appears for following:

1. with provider["registry.terraform.io/hashicorp/kubernetes"]
2. with module.helm.helm_release.cert-manager
3. with module.helm.helm_release.theia-cloud-crds[0]

Currently I am trying to understand and debug this error. Just though to share it here too, in case you have any information about this particular error.
